tests/Ionide.Analyzers.Tests/Suggestion/StructDiscriminatedUnionAnalyzerTests.fs

```diff
@@ -0,0 +1,155 @@
+module Ionide.Analyzers.Tests.Suggestion.StructDiscriminatedUnionAnalyzerTests
+
+open NUnit.Framework
+open FSharp.Compiler.CodeAnalysis
+open FSharp.Compiler.Text.Range
+open FSharp.Analyzers.SDK
+open FSharp.Analyzers.SDK.Testing
+open Ionide.Analyzers.Suggestion.StructDiscriminatedUnionAnalyzer
+
+let mutable projectOptions: FSharpProjectOptions = FSharpProjectOptions.zero
+
+[<SetUp>]
+let Setup () =
+    task {
+        let! opts = mkOptionsFromProject "net7.0" []
+        projectOptions <- opts
+    }
+
+[<Test>]
+let ``du without any field values`` () =
+    async {
+        let source =
+            """module Lib
+
+type Foo =
+    | Bar
+    | Barry
+    """
+
+        let ctx = getContext projectOptions source
+        let! msgs = structDiscriminatedUnionCliAnalyzer ctx
+        Assert.That(msgs, Is.Not.Empty)
+        let msg = msgs[0]
+        Assert.That(Assert.messageContains message msg, Is.True)
+    }
+
+[<Test>]
+let ``negative: du is already struct`` () =
+    async {
+        let source =
+            """module Lib
+
+[<Struct>]
+type Foo =
+    | Bar
+    | Barry
+    """
+
+        let ctx = getContext projectOptions source
+        let! msgs = structDiscriminatedUnionCliAnalyzer ctx
+        Assert.That(msgs, Is.Empty)
+    }
+
+[<Test>]
+let ``du with only primitive field values`` () =
+    async {
+        let source =
+            """module Lib
+
+type Foo =
+    | Bar of int
+    | Barry of float
+    | Bear of System.DateTime
+    | B4 of string
+    | B5 of System.TimeSpan
+    | B6 of int16 * int64 * uint * uint16 * byte * sbyte * float32 * decimal * char * bool
```

Comment:
There is a catch here: Tuple is class and not primitive. ValueTuple is a struct.

Reply:
Thanks, I didn't think about that.